components/compliance-service/inspec-agent/remote/remote.go

```diff
@@ -29,16 +33,28 @@ func RunSSMJob(ctx context.Context, ssmJob *types.InspecJob) *inspec.Error {
 	logrus.Debugf("Running ssm job: %+v", ssmJob)
 	// 1. CREATE TOKEN
 	// this is needed to hand over to inspec so it can report to automate
-	if RemoteJobInfo.TokensMgmtClient == nil {
+	if RemoteJobInfo.TokensMgmtClient == nil || RemoteJobInfo.PoliciesClient == nil {
 		logrus.Error("unable to create auth token")
 		return translateToInspecErr(fmt.Errorf("unable to connect to auth client: aborting job run for job %+v", ssmJob))
 	}
+	tokenID := fmt.Sprintf("inspec-to-automate-scanjob-%s", ssmJob.JobID)
+	ctx = auth_context.NewOutgoingContext(auth_context.NewContext(ctx, []string{"tls:service:deployment-service:internal"}, []string{}, "", ""))
```

Comment:
You are inadvertently masquerading as the deployment service here. That may be OK but there is a check for that in the gateway that is used for a specific purpose: https://github.com/chef/automate/blob/c8f110926945f0363759effd949e3cab1218fd62/components/automate-gateway/gateway/middleware/auth_interceptor.go#L246-L250

Probably better to use your own designator, perhaps:
```suggestion
	ctx = auth_context.NewOutgoingContext(auth_context.NewContext(ctx, []string{"tls:service:compliance-service:internal"}, []string{}, "", ""))
```

Reply:
ah yes! thank you!